lib/Common/Memory/Recycler.cpp

```diff
@@ -1320,11 +1320,22 @@ Recycler::LargeAlloc(HeapInfo* heap, size_t size, ObjectInfoBits attributes)
 {
     Assert((attributes & InternalObjectInfoBitMask) == attributes);
 
-    if (size >= HeapConstants::MaxLargeObjectSize)
+#if ENABLE_DEBUG_CONFIG_OPTIONS
+    size_t limit = GetRecyclerFlagsTable().MaxSingleAllocSizeInMB * 1024 * 1024;
+#else
+    size_t limit = CONFIG_FLAG(MaxSingleAllocSizeInMB) * 1024 * 1024;
```

Comment:
When would we not have a Recycler flags table? And how does this work for release builds? The flag is marked as FLAGNR?

Reply:
I don't want to change the default for free build before we have the data.
in free build it is a constant so preventing from corrupted